src/Accelerators/NNPA/Transform/ZLow/ZLowRewrite.cpp

```diff
@@ -33,6 +34,38 @@ using namespace mlir;
 namespace onnx_mlir {
 namespace zlow {
 
+/// Transform the zlow.reshape into a memref.reinterpret_cast as this pass
+/// operates after all memrefs are fully normalized, which is a requirement
+/// here.
+
+class ReshapeToReinterpretCastPattern : public OpRewritePattern<ZLowReshapeOp> {
+public:
+  using OpRewritePattern<ZLowReshapeOp>::OpRewritePattern;
+  LogicalResult matchAndRewrite(
+      ZLowReshapeOp reshapeOp, PatternRewriter &rewriter) const override {
+    Location loc = reshapeOp.getLoc();
+    MultiDialectBuilder<MemRefBuilder> create(rewriter, loc);
+    IndexExprScope currScope(&rewriter, loc);
+    // Here, cannot use the shape found in the reshape op, as it is the original
+    // shape before memref normalization.
+    Value input = reshapeOp.getX();
```

Comment:
We should check if X is normalized or not before processing further, by using something like this:
```
    // Input must have no affine layout. In other words, it has been normalized.
    if (hasNonIdentityLayout(input.getType()))
      return failure();
```

Without this check, I see that, in your lit test `zlow-rewrite.mlir`, `zlow.reshape` with affine_maps is still lowered to `memref.reinterpret_cast`

Reply:
Got it, I did not realize that `zlow-rewrite` ran twice. Its now fixed.